### ISSUE TITLE
Python Unicode Errors

### ISSUE THREAD

**Issue by danyeaw** [mintty/mintty#855]
If I run:
```
python3
>>> print("\uFFE6")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:/msys64/mingw64/lib/python3.7\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\uffe6' in position 0: character maps to <undefined>
```
Others are getting this same issue here https://github.com/pybee/briefcase/issues/179
I do not get this error when running mingw64 python3 from PowerShell:
```
Python 3.7.2 (default, Dec 25 2018, 05:36:22)  [GCC 8.2.1 20181214 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> print("\uFFE6")
￦
```
My PowerShell font doesn't allow me to see the actual W character, but I don't get the unicode error.

I tried all the tips on stack overflow for unicode errors on Windows consoles that I could find:
https://stackoverflow.com/questions/5419/python-unicode-and-the-windows-console

**Comment by Biswa96** [comment]
The file path looks weird -- mixing of forward & back slashes.

**Comment by danyeaw** [comment]
@Biswa96 I agree 😄

**Comment by BrianInglis** [comment]
Python problem not mintty problem - mintty displays: 
$ fgrep -a FFE6 src/charsets/unicode-symbols.txt
￦ U+0FFE6 FULLWIDTH WON SIGN

**Comment by danyeaw** [comment]
@BrianInglis Thanks. So I raised it here, because the same version of mingw64 python 3 works fine in a different console. So do you think it is most likely something wrong with the packaging of Python 3 for MSYS2 that is conflicting with mintty?

**Comment by BrianInglis** [comment]
With my fonts and fallbacks, mintty displays the correct glyph: 
$ fgrep -a FFE6 src/charsets/unicode-symbols.txt
￦ U+0FFE6 FULLWIDTH WON SIGN
$ fgrep -a FFE6 src/charsets/unicode-symbols.txt | cat -A
M-oM-?M-& U+0FFE6 FULLWIDTH WON SIGN$
for comparison, bash printf also outputs different char codes, with leading M-m instead of M-o: 
$ printf "\uFFE6\t\U0000FFE6\n"
퟿�      ퟿�
$ printf "\uFFE6\t\U0000FFE6\n" | cat -A
M-mM-^_M-?M-mM-?M-&^IM-mM-^_M-?M-mM-?M-&$
pipe thru cat -A to see what char codes are generated in your consoles: check your selected charset is Unicode and console fonts selected or fallbacks provide that glyph, or the glyphs you want to use.

**Comment by BrianInglis** [comment]
Just noticed the error message shows Python is complaining because it is using Windows encoding cp1252 not Unicode: "C:/msys64/mingw64/lib/python3.7\encodings\cp1252.py": check your locale and related encodings, and those used by Python.

**Comment by mintty** [comment]
> My PowerShell font doesn't allow me to see the actual ￦ character, but I don't get the unicode error.

That's because the "conhost" console uses completely different mechanisms to output Unicode characters. If the application uses the UTF-16 API output will always work but not in a terminal where proper encoding settings are necessary.

**Comment by mintty** [comment]
Duplicate of #846, by the way. You may try `winpty` as a workaround.
Note that the new Windows ConPTY API might support to craft a built-in workaround for the situation, in cygwin, as discussed in #804. Provided that some eager person who understands this intricate API is willing to undergo the torture to deal with it...

**Comment by Alexander-Wilms** [comment]
The `winpty` workaround worked for me on Windows 10 with mintty 3.6.0 shipped with Git for Windows.

**Comment by vtgn** [comment]
Hi!
I've got exactly the same problem with my git bash, which uses mintty. See my comment here: https://github.com/git-for-windows/git/issues/1254

I've changed absolutely nothing with my git bash, python, and python dependencies. However, the problem has happened suddenly, it wasn't present 2 weeks ago.

So I would like to understand what could have happened. The python installation is exactly the same, I haven't reinstalled it. And it works perfectly from the windows cmd console. So why suddenly python uses cp1252 on my git bash? My git-bash uses the system configuration for locale, which is correct:

> $ locale
LANG=
LC_CTYPE="fr_FR.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_ALL=

And I've never configured anything on git bash for python.
The problem is resolved if I set PYTHONIOENCODING with 1, but I've never needed it before. I just would like to understand what could have happened.

I noticed too in my comment at the given link a testimony of someone having the same problem that has been resolved by a reinstallation of mintty.
